### DISCUSSION TITLE
PiFireOS

### DISCUSSION THREAD

**Discussion by calonmerc** [nebhead/PiFire#128] [General]
As mentioned, I'm going to work on creating a PiFireOS image builder for Raspberry Pi images using [CustomPiOS](https://github.com/guysoft/CustomPiOS) and Github actions to create nightly builds. 

I had a couple of questions about the usability/install process:
- Currently everything is installed to `/usr/local/bin`. Is there any particular reasoning behind it, instead of just in the users home directory? On my Raspberry Pi, I moved everything to the `/home/pi` directory then changed the paths in supervisor for the control and webapp. 
- Currently on auto-install, the user and password for Supervisor are set up, this won't be possible in an image environment. Can probably use the `/boot/userconf.txt` file that should set the password for an initial user, but I'm not sure how/if CustomPiOS handles that
- I'm not entirely sure if we can maintain the /tmp as RAM based with the image based scripts, but I will look into it for sure. I believe there's a section to run commands after first boot, so it might be in there.

**Comment by nebhead** [comment]
>     * Currently everything is installed to `/usr/local/bin`. Is there any particular reasoning behind it, instead of just in the users home directory? On my Raspberry Pi, I moved everything to the `/home/pi` directory then changed the paths in supervisor for the control and webapp.

Yes, the reason we moved to /usr/local/bin was to allow the installation to be generic, and not rely on a pi user.  It was reported in several articles, the default pi user is going to be deprecated in future Raspbian OSes.  Thus, it was prudent to remove the dependency of a user account.  
 
>     * Currently on auto-install, the user and password for Supervisor are set up, this won't be possible in an image environment. Can probably use the `/boot/userconf.txt` file that should set the password for an initial user, but I'm not sure how/if CustomPiOS handles that

That's fine, it's not critical to have these setup initially.  As you mention in the next question, we could also have a post-install script that runs and does the necessary installation steps that can't be done from a base install.  

>     * I'm not entirely sure if we can maintain the /tmp as RAM based with the image based scripts, but I will look into it for sure. I believe there's a section to run commands after first boot, so it might be in there.

Hopefully we could run some kind of post-installation script that would be able to configure these things.  PiFire could potentially trigger this if needed.

**Comment by calonmerc** [comment]
> Yes, the reason we moved to /usr/local/bin was to allow the installation to be generic, and not rely on a pi user. It was reported in several articles, the default pi user is going to be deprecated in future Raspbian OSes. Thus, it was prudent to remove the dependency of a user account.

Ahh that makes sense. In the image environment we could automatically make that user, but it would/should make sense to be agnostic.... and the image should match the installer if at all possible, so we will make it match!

> That's fine, it's not critical to have these setup initially. As you mention in the next question, we could also have a post-install script that runs and does the necessary installation steps that can't be done from a base install.

As far as I know, the same from above would dictate we shouldn't use a standard user/password, but these would be set on first boot (not first log in), so unless a user changes them, should the UI be defaulted to off? Or could/should there be an initial setup wizard built into the webapp? (I would prefer to keep them as separated as possible, such that the image just runs either the latest stable or latest dev branch and has no requirements from the PiFire codebase)

FWIW I had this typed within an hour of your reply, but I got distracted by something shiny.

**Reply by nebhead** [reply]
> Ahh that makes sense. In the image environment we could automatically make that user, but it would/should make sense to be agnostic.... and the image should match the installer if at all possible, so we will make it match!

Sounds good! 
 
> As far as I know, the same from above would dictate we shouldn't use a standard user/password, but these would be set on first boot (not first log in), so unless a user changes them, should the UI be defaulted to off? Or could/should there be an initial setup wizard built into the webapp? (I would prefer to keep them as separated as possible, such that the image just runs either the latest stable or latest dev branch and has no requirements from the PiFire codebase)

Actually, yeah, it could be built into the webapp such that we just do the necessary configuration on first boot.  

> FWIW I had this typed within an hour of your reply, but I got distracted by something shiny.

LOL.  I'm no hurry here, take all the time you want. :smile:
